heartbeat/security/security.go

```diff
@@ -97,5 +98,10 @@ func setCapabilities() error {
 		return fmt.Errorf("error setting new process capabilities via setcap: %w", err)
 	}
 
+	_, err = cap.Prctl(unix.PR_SET_DUMPABLE, 1)
+	if err != nil {
+		return fmt.Errorf("error setting dumpable flag via prctl: %w", err)
+	}
```

Comment:
Perhaps move this to a separate function, so it's clear that it's unrelated to the capabilities changes above? And add a comment explaining that it's needed because the `heartbeat` binary has elevated file capabilities set?

Reply:
Probably also worth adding a comment why this is important, i.e. so elastic-agent can query heartbeat's metrics.